doc/source/tutorial/stats/rv_infrastructure.md

```diff
@@ -0,0 +1,920 @@
+
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+x = np.linspace(eps, 15, 300)
+
+c, scale = res_mle.x
+X = Weibull(c=c)*scale
+plt.plot(x, X.pdf(x), '-', label='numerical optimization')
+
+c, loc, scale = params
+Y = Weibull(c=c)*scale
+plt.plot(x, Y.pdf(x), '--', label='`scipy.stats.fit`')
+
+plt.hist(data, bins=np.linspace(0, 20, 30), density=True, alpha=0.1)
+plt.xlabel('x')
+plt.ylabel('pdf(x)')
+plt.legend()
+plt.ylim(0, 0.5)
+plt.show()
+```
+
+However, with this approach, it is trivial to make changes to the fitting procedure to suit our needs, enabling estimation procedures other than those provided by [`rv_continuous.fit`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_continuous.fit.html#scipy.stats.rv_continuous.fit). For instance, we can perform [maximum spacing estimation](https://en.wikipedia.org/wiki/Maximum_spacing_estimation) (MSE) by changing the objective function as follows.
+
+```{code-cell} ipython3
+def nlps(params):
+    c, scale = params
+    X = Weibull(c=c) * scale
+    x = np.sort(np.concatenate((data, X.support())))  # Append the endpoints of the support to the data
+    return -X.logcdf(x[:-1], x[1:]).sum().real  # Minimize the sum of the logs the probability mass between points
+
+res_mps = optimize.minimize(nlps, x0, bounds=bounds)
+res_mps.x
+```
+
+For method of [L-moments](https://en.wikipedia.org/wiki/L-moment) (which attempts to match the distribution and sample L-moments):
+
+```{code-cell} ipython3
+def lmoment_residual(params):
+    c, scale = params
+    X = Weibull(c=c) * scale
+    E11 = stats.order_statistic(X, r=1, n=1).mean()
+    E12, E22 = stats.order_statistic(X, r=[1, 2], n=2).mean()
+    lmoments_X = [E11, 0.5*(E22 - E12)]  # the first two l-moments of the distribution
+    lmoments_x = stats.lmoment(data, order=[1, 2])  # first two l-moments of the data
+    return np.linalg.norm(lmoments_x - lmoments_X)  # Minimize the norm of the difference
+
+x0 = [0.4, 3]  # This method is a bit sensitive to the initial guess
+res_lmom = optimize.minimize(lmoment_residual, x0, bounds=bounds)
+res_lmom.x, res_lmom.fun  # residual should be ~0
+```
+
+The plots for all three methods look similar and seem to describe the data, giving us some confidence that we have a good fit.
+
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+x = np.linspace(eps, 10, 300)
+
+for res, label in [(res_mle, "MLE"), (res_mps, "MPS"), (res_lmom, "L-moments")]:
+    c, scale = res.x
+    X = Weibull(c=c)*scale
+    plt.plot(x, X.pdf(x), '-', label=label)
+
+plt.hist(data, bins=np.linspace(0, 10, 30), density=True, alpha=0.1)
+plt.xlabel('x')
+plt.ylabel('pdf(x)')
+plt.legend()
+plt.ylim(0, 0.3)
+plt.show()
+```
+
+Estimation does not always involve fitting distributions to data. For instance, in gh-12134, a user needed to calculate the shape and scale parameters of the Weibull distribution to achieve a given mean and standard deviation. This fits easily into the same framework.
```

Comment:
`gh-12134` does not automatically create an issue-link; we should make it clickable

Reply:
Sure, I forgot that one.